packages/flutter/lib/src/material/menu_anchor.dart

```diff
@@ -3029,6 +3044,10 @@ class _MenuItemLabel extends StatelessWidget {
           ),
       ],
     );
+    if (semanticsLabel != null) {
+      menuItemLabel = Semantics(label: semanticsLabel, excludeSemantics: true, child: menuItemLabel);
```

Comment:
Do we want to exclude semantics for all of the children, or just the `widget.child`?  I can see someone wanting to put some kind of button in as part of the trailing or leading icon, and those would want to keep their semantics. (See the "Zoom" line in the Chrome "..." menu for an example).

We could also just assume that people who want to do something that complex will be making their own widget to add to the menu, but they'll probably start by trying to use this, so if it's possible to let them, then we might as well not prevent it.

Reply:
Or I suppose they could still do it, they'd just have to not supply a semantics label (and know not to do that).

Reply:
That's interesting, I can definitely see that scenario playing out. In our case,  we need to overwrite the semantic label on both the child and the serialized shortcut decorations, while still displaying the generated shortcut icons. 

What if instead of passing in the label, we exposed a field called `optionalSemanticsProperties` and said explicitly in the documentation `Setting these properties overwrites the Semantics of the entire Widget`, or something similar? That seems to me to be a clearer message of "this applies to all descendants" than just a label would.

Reply:
Yes, that might work.  But what would that pass in?  Not just a string, right?

 Maybe a better idea would be to have an optional `semanticsBuilder` argument that is a callback that takes a `Widget` and returns a `Widget`. The input to the builder is the child for the semantics widget (`menuItemLabel` in your change), and the output is a `Semantics` widget wrapping the given child. It shouldn't require a `Semantics` widget return value, since there are cases where you might want to wrap it in something else (e.g. an `ExcludeSemantics` widget).

Then you have full control over what the semantics looks like for the widget.

However, at that point, why can't you just wrap the `MenuItemButton` with a `Semantics` widget and get what you want? I seem to vaguely remember discussing this, but I can't remember why it didn't work.

Reply:
I also might not remember exactly, but I think the problem was mostly with SubmenuButton, since wrapping it in a Semantics label resulted in merging the submenu semantics, or replacing them if I'm not mistaken. At a MenuItemButton level, I'm not sure if that doesn't work. I'd need to test that again

Reply:
Ahh, yes, that sounds familiar.  I would think that the problem has to do with only the `SubmenuButton`, since that would definitely cause the children's semantics to be lumped together.  You should be able to apply a `Semantics` widget around each child to override their semantics, but I think we even talked about that and decided that was tedious, and we arrived here.  My callback suggestion would definitely get us back to tedious again.

> What if instead of passing in the label, we exposed a field called `optionalSemanticsProperties` and said explicitly in the documentation `Setting these properties overwrites the Semantics of the entire Widget`, or something similar? That seems to me to be a clearer message of "this applies to all descendants" than just a label would.

How about if we go back to this, except I'd call it `semanticsLabel` as before, since making it nullable should imply optional anyhow, and it's just the label, not the other properties. Make clear that it's just the label in the docs too, and maybe mention that if they want to override other semantics properties, they can wrap the item in a `Semantics` widget.

Reply:
Cool, so I can update the docs here to let the user know they can wrap this in Semantics. But I'm a little confused about the difference between this proposal and the existing PR? It seems like that's what this code is doing. The main addition is the extra `excludeSemantics` field which is there to prevent the Screen Reader from reading beyond what was given in the label. Beyond that, which part of the PR should I tweak?

Reply:
Sorry, I wasn't too clear there. I guess this PR is doing what I was I was asking for, except for the docs.